src/transaction/build.rs

```diff
@@ -158,107 +172,150 @@ pub struct BuildTransactionRequestForLocalSigningFuture<T: Transport> {
     /// `Option` is used here as the `Future` implementation requires moving the
     /// transaction options in order to construct the `TransactionRequest`.
     options: Option<TransactionRequestOptions>,
-    /// The inner future for retrieving the list of accounts on the node to
-    /// determine the default account if none were specified.
+    /// The inner future for retrieving the missing parameters required for
+    /// finalizaing the transaction request: the list of accounts on the node
+    /// (in order determine the default account if none was specified) and the
+    /// resolved gas price (in case a scaled gas price is used).
     #[pin]
-    state: LocalAccountState<T>,
-}
-
-/// The inner state of the future for building locally signed transactions.
-#[must_use = "futures do nothing unless you `.await` or poll them"]
-#[pin_project]
-enum LocalAccountState<T: Transport> {
-    /// The account to use for local signing is already known.
-    Address(Address),
-    /// Retrieving the list of accounts that can be used for local signing from
-    /// the node.
-    RetrievingAccounts(#[pin] CompatCallFuture<T, Vec<Address>>),
+    params: LocalParamsFuture<T>,
 }
 
 impl<T: Transport> BuildTransactionRequestForLocalSigningFuture<T> {
     /// Create a new future for building a locally singed transaction request
     /// from a partial transaction object and account information.
-    pub fn new(web3: &Web3<T>, from: Option<Address>, options: TransactionRequestOptions) -> Self {
+    pub fn new(
+        web3: &Web3<T>,
+        from: Option<Address>,
+        gas_price: GasPrice,
+        options: TransactionRequestOptions,
+    ) -> Self {
         let options = Some(options);
-        let state = if let Some(from) = from {
-            LocalAccountState::Address(from)
-        } else {
-            LocalAccountState::RetrievingAccounts(web3.eth().accounts().compat())
+        let params = {
+            let eth = web3.eth();
+            let accounts = maybe!(from.map(|from| vec![from]), eth.accounts().compat());
+            let gas_price = gas_price.resolve_for_transaction_request(web3);
+            future::join(accounts, gas_price)
         };
 
-        BuildTransactionRequestForLocalSigningFuture { options, state }
+        BuildTransactionRequestForLocalSigningFuture { options, params }
     }
 }
 
 impl<T: Transport> Future for BuildTransactionRequestForLocalSigningFuture<T> {
     type Output = Result<TransactionRequest, ExecutionError>;
 
-    #[project]
     fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         let mut this = self.project();
 
-        #[project]
```

Comment:
I guess we don’t need this “project” tag-thing anymore?

Reply:
Not in this particular place. It’s still used other places. It’s only needed when doing pin projections on enums